src/python/pants/backend/cc/subsystems/compiler.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import logging
+
+from pants.core.util_rules.external_tool import TemplatedExternalTool
+from pants.option.option_types import StrListOption, StrOption
+from pants.option.subsystem import Subsystem
+from pants.util.strutil import softwrap
+
+logger = logging.getLogger(__name__)
+
+
+class CCOptions:
```

Comment:
@tdyas As discussed in the other PR, these should probably be `cxx` instead of `cpp` (in order to not get confused with the pre-processor flags), and then maybe even add the pre-processor flags uniquely.

Reply:
See https://github.com/pantsbuild/pants/discussions/16777